test/network.test.js

```
@@ -0,0 +1,98 @@
+  const successes = new Set([1, 3]);
+  function FakeImage() {
+    this.onload = null;
+    this.onerror = null;
+  }
+  Object.defineProperty(FakeImage.prototype, "src", {
+    set(v) {
+      const m = v.match(/192\.168\.0\.(\d+)/);
+      const idx = m ? parseInt(m[1], 10) : 0;
+      queueMicrotask(() => {
+        if (successes.has(idx)) {
+          this.onload && this.onload();
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)